### ISSUE TITLE
Write something about mocking HTTP services

### ISSUE THREAD

**Issue by Mikilll94** [lmiller1990/vue-testing-handbook#170]
I have read the whole handbook. It is great, I appreciate it 👍  

However, IMO it lacks information about mocking HTTP services. Currently, I noticed that you only describe how to mock the whole `axios` module which is fine for demonstration purposes. However, in a real-world application, you don't want to do this. Instead, you extract your HTTP communication to separate files most-often called "services", i.e. accountService, taxService, etc. And you mock these services, not whole `axios`.

This is a better approach because in real-world applications you often have components with multiple HTTP calls, some logic around them, and some error handling.

And the solution like this is not fine:
![Screenshot 2020-09-13 at 14 18 16](https://user-images.githubusercontent.com/17653648/93017830-fb35e180-f5cb-11ea-999d-91c3b39ff2d8.png)

because you mock all axios "GET" methods with the same response.

IMO, for mocking HTTP services, it would be nice to introduce ES6 class mocks from Jest, but this is just my proposition.

**Comment by lmiller1990** [comment]
Yep, I would like to add this for sure. I think we need two sections:

1. add an entire section about jest mocks
2. add something small about mocking axios (or services, or whatever) on the page that talks about mocking axios. It could link to (1)

There is actually a placeholder for this (last page of the handbook, about Jest mocks).

Are either of these something you are interested in contributing?
